clouddrift/analytic.py

```diff
@@ -0,0 +1,80 @@
+"""
+This module provides signal processing functions.
+"""
+
+import numpy as np
+from typing import Optional, Tuple, Union
+import xarray as xr
+import pandas as pd
+import warnings
+
+
+def anatrans(
+    x: Union[list, np.ndarray, xr.DataArray, pd.Series],
+    boundary: Optional[str] = "mirror",
+) -> np.ndarray:
+    """returns the analytic part of a real-valued signal or of a complex-valued
+    signal. To obtain the anti-analytic part of a complex-valued signal apply anatrans
+    to the conjugate of the input. Anatrans removes the mean of the input signals.
+
+    Parameters
+    ----------
+    x : np.ndarray
+    boundary : str, optional ["mirror", "zeros", "periodic"]
+
+    Returns
+    -------
+    z : np.ndarray
+
+    Examples
+    --------
+
+    To obtain the analytic part of a real-valued signal
+    >>> x = np.random.rand(99)
+    >>> z = anatrans(x)
+
+    To obtain the analytic and anti-analytic parts of a complex-valued signal
+    >>> z = np.random.rand(99)+1j*np.random.rand(99)
+    >>> zp = anatrans(z)
+    >>> zn = anatrans(np.conj(z))
+
+    Raises
+    ------
+    ValueError
+        If ``boundary not in ["mirror", "zeros", "periodic"]``.
+    """
+    # assume unidimensional input; add dimension option
+    m0 = len(x)
+
+    # remove mean
+    x = x - np.mean(x)
```

Comment:
Both work but this form is more Pythonic.

```suggestion
    x -= np.mean(x)
```